### PR TITLE
first run at testing, these can currently assume a location in a mage…

### PR DESCRIPTION
**Pull request by bjwest55**
…nto 2 installation. But its a start

### CODE REVIEW

**Comment by julesbravo** [comment]
@bjwest55 What do you need from me here?

**Comment by bjwest55** [comment]
@julesbravo I guess nothing. This kind of signals the boost team taking ownership. Didn't know if you still wanted to be aware of this code as it changes. If not I can pretty much just keep the PRs to other eng members

File: Helper/Generator.php [hunk 1]
```diff
@@ -476,17 +468,17 @@ protected function addOptionsToRecord($product) {
...
+                        Generator::setRecordValue($productRecord,$field, $value->getTitle(),$this->ignoreFields);
                     }
                 }
             }
         }
     }
 
-    protected function addImagesToRecord($product) {
-        $this->setRecordValue('cached_thumbnail', $this->getThumbnail($product));
+    protected function addImagesToRecord($product, &$productRecord) {
```
Comment (evanjohnso) [inline Helper/Generator.php:478]: what's this `&$` syntax

File: Helper/Generator.php [hunk 2]
```diff
@@ -530,13 +522,13 @@ protected function getThumbnail($product, $type = 'product_thumbnail_image', $im
         return $resizedImage;
     }
 
-    protected function addStockInfoToRecord($product) {
+    protected function addStockInfoToRecord($product,&$productRecord) {
         $stockItem = $this->stockRegistryInterface->getStockItem($product->getId());
-        $this->setRecordValue('in_stock', $stockItem->getIsInStock());
-        $this->setRecordValue('stock_qty', $stockItem->getQty());
+        Generator::setRecordValue($productRecord,'in_stock', $stockItem->getIsInStock(),$this->ignoreFields);
```
Comment (evanjohnso) [inline Helper/Generator.php:527]: is `::` for static methods ?

**Review by evanjohnso — APPROVED** [review]

**Comment by evanjohnso** [comment]
i'm sure you're already on it, but if ya got tests, prolly want them to run in a GitHub actions else they're prolly not very useful 😆

File: Helper/Generator.php (hunk 1 above)
Comment (bjwest55) [reply Helper/Generator.php:478]: you're passing the product by reference. so the value going in can change

File: Helper/Generator.php (hunk 2 above)
Comment (bjwest55) [reply Helper/Generator.php:527]: yes

**Comment by julesbravo** [comment]
@evanjohnso @bjwest55 - I'm a bit confused. It was my understanding last that the DACQs team was going to be rewriting this. I had a discussion with them about the module earlier this month. Who is owning this moving forward?

**Pull request merged**